server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aeternity/rpc/AeternityRPCClient.java

```diff
@@ -0,0 +1,1087 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.generalbytes.batm.common.currencies.CryptoCurrency;
+import com.generalbytes.batm.server.extensions.extra.aeternity.AeternityWallet;
+import com.generalbytes.batm.server.extensions.extra.common.RPCClient;
+import com.kryptokrauts.aeternity.generated.model.Account;
+import com.kryptokrauts.aeternity.generated.model.GenericSignedTx;
+import com.kryptokrauts.aeternity.generated.model.GenericTx;
+import com.kryptokrauts.aeternity.generated.model.KeyBlock;
+import com.kryptokrauts.aeternity.generated.model.PostTxResponse;
+import com.kryptokrauts.aeternity.generated.model.SpendTx;
+import com.kryptokrauts.aeternity.generated.model.Tx;
+import com.kryptokrauts.aeternity.generated.model.UnsignedTx;
+import com.kryptokrauts.aeternity.sdk.service.ServiceConfiguration;
+import com.kryptokrauts.aeternity.sdk.service.account.AccountService;
+import com.kryptokrauts.aeternity.sdk.service.account.AccountServiceFactory;
+import com.kryptokrauts.aeternity.sdk.service.chain.ChainService;
+import com.kryptokrauts.aeternity.sdk.service.chain.ChainServiceFactory;
+import com.kryptokrauts.aeternity.sdk.service.transaction.TransactionService;
+import com.kryptokrauts.aeternity.sdk.service.transaction.TransactionServiceConfiguration;
+import com.kryptokrauts.aeternity.sdk.service.transaction.TransactionServiceFactory;
+import com.kryptokrauts.aeternity.sdk.service.transaction.type.AbstractTransaction;
+import com.kryptokrauts.aeternity.sdk.service.transaction.type.impl.SpendTransaction;
+
+import si.mazi.rescu.HttpStatusIOException;
+import si.mazi.rescu.RestProxyFactory;
+import wf.bitcoin.javabitcoindrpcclient.GenericRpcException;
+
+import wf.bitcoin.krotjson.HexCoder;
+
+public class AeternityRPCClient extends RPCClient /* extends BitcoinJSONRPCClient */ {
+    private static final Logger log = LoggerFactory.getLogger("batm.master.extensions.RPCClient");
+    //private String cryptoCurrency = CryptoCurrency.AE.getCode();
+    private final AeternityV2API api;
+    
+    
+    final static String baseUrl = "https://sdk-mainnet.aepps.com/v2"; // default: https://sdk-mainnet.aepps.com/v2
+    final com.kryptokrauts.aeternity.sdk.constants.Network mainnet = com.kryptokrauts.aeternity.sdk.constants.Network.MAINNET; 
+    private static ServiceConfiguration serviceConf;
+    private static AccountService accountService;
+    private static ChainService chainService;
+    private static TransactionService transactionService;
+    private static BigDecimal divider = new BigDecimal("1000000000000000000");
+    
+    public AeternityRPCClient(String rpcUrl) throws MalformedURLException {
+    	super(CryptoCurrency.AE.getCode(), baseUrl);
+        if (rpcUrl == null) rpcUrl = baseUrl;
+        api = RestProxyFactory.createProxy(AeternityV2API.class, "https://roma-net.mdw.aepps.com");// aeternity official block explorer api, https://explorer.aepps.com
```

Comment:
Please don't use centralized service to track transactions. Please use wallet to provide you this information, just as with BCH.

Reply:
Now im using external public node, hosted by Aeternity. Will make tests with local node and write back